service/service.go

```diff
@@ -179,15 +173,16 @@ func New(config Config) (*Service, error) {
 			K8sClient: k8sClient,
 			Logger:    config.Logger,
 
-			Azure:            azure,
-			AzureConfig:      azureConfig,
-			Ignition:         Ignition,
-			OIDC:             OIDC,
-			InstallationName: config.Viper.GetString(config.Flag.Service.Installation.Name),
-			ProjectName:      config.ProjectName,
-			RegistryDomain:   config.Viper.GetString(config.Flag.Service.RegistryDomain),
-			SSOPublicKey:     config.Viper.GetString(config.Flag.Service.Tenant.SSH.SSOPublicKey),
-			VMSSCheckWorkers: config.Viper.GetInt(config.Flag.Service.Azure.VMSSCheckWorkers),
+			Azure:                          azure,
+			CPAzureClientCredentialsConfig: cpAzureClientCredentialsConfig,
+			CPSubscriptionID:               config.Viper.GetString(config.Flag.Service.Azure.SubscriptionID),
```

Comment:
This is the key. We decouple the credentials used from the subscription where we will manage resources.